projects/step-frontend/src/lib/modules/_common/components/is-used-by-list/is-used-by-list.component.ts

```diff
@@ -0,0 +1,90 @@
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+import {
+  AJS_MODULE,
+  AugmentedAdminService,
+  Mutable,
+  ProjectDto,
+  ReferencesService,
+  TableLocalDataSource,
+} from '@exense/step-core';
+import { downgradeComponent, getAngularJSGlobal } from '@angular/upgrade/static';
+import { BehaviorSubject, switchMap, shareReplay, tap, pipe, map } from 'rxjs';
+
+type SearchType = 'PLAN_ID' | 'PLAN_NAME' | 'KEYWORD_ID' | 'KEYWORD_NAME' | 'RESOURCE_ID' | 'RESOURCE_NAME';
+type InProgress = Mutable<Pick<IsUsedByListComponent, 'inProgress'>>;
+
+@Component({
+  selector: 'step-is-used-by-list',
+  templateUrl: './is-used-by-list.component.html',
+  styleUrls: ['./is-used-by-list.component.scss'],
+})
+export class IsUsedByListComponent {
+  @Input() title!: string;
+  @Input() type!: string;
+  @Input() id!: string;
+
+  @Output() onClose = new EventEmitter<any>();
+
+  readonly inProgress: boolean = false;
+
+  projectIdToDtoMap: { [id: string]: ProjectDto } = {};
+  emptyResults: boolean = false;
+
+  private _findReferencesRequest$ = new BehaviorSubject<unknown>({});
+  readonly references$ = this._findReferencesRequest$.pipe(
+    tap(() => ((this as InProgress).inProgress = true)),
+    switchMap(() =>
+      this._referencesService.findReferences({
+        searchType: this.type as SearchType,
+        searchValue: this.id,
+      })
+    ),
+    tap((result) => {
+      this.emptyResults = result.length === 0;
+      if (!this.emptyResults) {
+        const firstResultHasProject = !!result[0].attributes!['project'];
```

Comment:
We should replace it with: 

```suggestion
        const firstResultHasProject = !!result[0].attributes?.['project'];
```

Reply:
I see we are using this attribute with guarantee in other places too so it doesn't make sense to replace it here. In general I don't like guaranteeing variables. Check if we can move the others to optional (?), if not this can stay.

Reply:
this code has been removed anyways due to loading of tenant from rootScope